### PR TITLE
Add p3a for set as default

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/31056

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/misc_metrics/misc_android_metrics.h [hunk 1]
```diff
@@ -15,6 +15,9 @@ class SearchEngineTracker;
 
 namespace misc_metrics {
 
+inline constexpr char kBraveCoreIsDefaultHistogramName[] =
```
Comment (DJAndries) [inline browser/misc_metrics/misc_android_metrics.h:18]: if we have no plans to use this constant externally, then perhaps we should just move it into an anonymous namespace within `misc_android_metrics.cc`

**Review by DJAndries — APPROVED** [review]

File: android/java/org/chromium/chrome/browser/set_default_browser/BraveSetDefaultBrowserUtils.java [hunk 2]
```diff
@@ -64,6 +63,22 @@ public static boolean isBraveSetAsDefaultBrowser(Context context) {
...
     }
 
+    public static boolean isAppSetAsDefaultBrowser(Context context) {
+        Intent browserIntent =
+                new Intent(Intent.ACTION_VIEW, Uri.parse(UrlConstants.HTTP_URL_PREFIX));
+        ResolveInfo resolveInfo =
+                context.getPackageManager()
+                        .resolveActivity(
+                                browserIntent,
+                                supportsDefault() ? PackageManager.MATCH_DEFAULT_ONLY : 0);
```
[the remark this answers is not in the document]
Comment (stoletheminerals) [reply android/java/org/chromium/chrome/browser/set_default_browser/BraveSetDefaultBrowserUtils.java:73]: this doesn't launch any components, just compare two package names. sgtm

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Review by tapanmodh — APPROVED** [review]
LGTM

**Comment by GeetaSarvadnya** [comment]
Verification **PASSED** on `Vivo X70 Pro version 12` running `Bravemonoarm64.apk_1.67.1`

<details>
<summary>Brave.Core.IsDefault = 0_PASSED</summary>

1. Clean profile 1.67.1
2. Make sure brave is not default browser in Android device
3. Open brave://local-state file
4. Look for `Brave.Core.IsDefault` merics
5. Confirmed that the metrics `Brave.Core.IsDefault` response value is `0` as expected as brave is not default browser

Screenshot_20240411_112632

</details>


<details>
<summary>Brave.Core.IsDefault = 1_PASSED</summary>

1. Clean profile 1.67.1
2. Set brave as default browser via Hamburger menu option
3. Make sure Brave is the default browser via device settings menu
4. Restart the browser
5. Open brave://local-state file
6. Look for `Brave.Core.IsDefault` merics
7. Confirmed that the metrics `Brave.Core.IsDefault` response value is `1` as expected as brave is set as default browser

Example | Example
----------|---------
Screenshot_20240411_113054 | Screenshot_20240411_113111


</details>

**Pull request merged**